modules/geo-layers/src/terrain-layer/terrain-layer.js

```diff
@@ -168,6 +180,16 @@ export default class TerrainLayer extends CompositeLayer {
     });
   }
 
+  // Update zRange of viewport
+  onViewportLoad(data) {
```

Comment:
Looks like this triggers an infinite loop:

- viewport changed -> get tiles based on the last `zRange`
- loaded -> update `zRange`
- get tiles based on the updated `zRange`
- loaded -> update `zRange`
- get tiles based on the updated `zRange`
- ...

I'd recommend only updating the state if `zRange` has changed deeply, or even, only updating it if the existing `zRange` does not contain the new one.